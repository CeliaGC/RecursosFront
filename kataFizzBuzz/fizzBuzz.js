for (let x = 1; x <= 100; x++) {  
  let result = '';
  if(x % 3 == 0){
    result += 'Fizz';
  }
  if(x % 5 == 0){
    result += 'Buzz';
  }
  console.log(result || x);
}

/* Línea 1: Se inicia un bucle for. Establece una variable x con un valor 
inicial de 1 y ejecutará el bucle siempre que x sea menor o igual a 100. 
Después de cada iteración, se incrementa el valor de x en 1 (x++).

Línea 2: Aquí se crea una variable llamada result y se le asigna una 
cadena vacía como valor inicial. Esta variable se utilizará para construir 
la salida de cada número o palabra.

Línea 3 y 4: Esta línea verifica si el valor de x es divisible por 3 
(es decir, si el restode la división es 0). Si es cierto, agrega la 
palabra 'Fizz' a la variable result.

Línea 5 y 6: Esta línea verifica si el valor  de x es divisible por 5. 
Si es cierto, agrega la palabra 'Buzz' a la variable result.

Línea 9: Aquí, se utiliza console.log() para imprimir en la consola el 
valor de result. Si result es una cadena no vacía debido a las condiciones 
anteriores (es decir, si se concatenó 'Fizz' o 'Buzz' a ella), entonces se 
imprime el contenido de result. Si result sigue siendo una cadena vacía, 
se imprimirá simplemente el valor de x.

Строка 1: Запускается цикл for. Он устанавливает переменную x с начальным значением 
и цикл будет выполняться всякий раз, когда x будет меньше или равно 100. 
После каждой итерации значение x увеличивается на 1 (x++).

Строка 2: Здесь создается переменная result и в качестве начального значения ей присваивается пустая строка. 
пустую строку в качестве начального значения. Эта переменная будет использоваться для построения 
вывода каждого числа или слова.

Строки 3 и 4: В этой строке проверяется, делится ли значение x на 3 
(т.е. если остаток от деления равен 0). Если это так, то к результату прибавляется 
слово 'Fizz' в переменную result.

Строки 5 и 6: В этой строке проверяется, делится ли значение x на 5. 
Если верно, то в переменную result добавляется слово 'Buzz'.

Строка 9: Здесь console.log() используется для вывода на консоль значения 
значение результата. Если результат является непустой строкой в силу вышеприведенных условий (т.е. при конкатенации 
выше (т.е. если к ней было конкатенировано 'Fizz' или 'Buzz'), то выводится содержимое result. 
выводится содержимое result. Если результат по-прежнему является пустой строкой, 
то будет выведено просто значение x.
*/