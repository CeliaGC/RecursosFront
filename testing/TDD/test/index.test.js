import fizzBuzz from "../src";


test('devuelve fizz si es divisible entre 3', () => {
 //Arrange
let number = 3;

//Act
let result = fizzBuzz(number);

//Assert
expect(result).toBe("fizz");

});

test('devuelve buzz si el número es divisible entre 5', ()=> {
    let number = 5;

    let result = fizzBuzz(number);
    
    expect(result).toBeTruthy();
    expect(result).toBe("buzz");
});

test('devuelve buzz si el número es divisible entre 5', ()=> {
    let number = 15;

    let result = fizzBuzz(number);
    
    expect(result).toBeTruthy();
    expect(result).toBe("fizzBuzz");
});

test('devuelve números del 1 al 100 con las condiciones anteriores',()=>{

    let numbers = Array.from({length: 100}, (_, i) => i + 1);
  

    let result = fizzBuzz(numbers);
    

    expect(result).toMatchSnapshot();

})





