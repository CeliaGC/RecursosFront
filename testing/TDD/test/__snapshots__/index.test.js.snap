// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`devuelve n√∫meros del 1 al 100 con las condiciones anteriores 1`] = `
[
  1,
  2,
  "fizz",
  4,
  "buzz",
  "fizz",
  7,
  8,
  "fizz",
  "buzz",
  11,
  "fizz",
  13,
  14,
  "fizzBuzz",
  16,
  17,
  "fizz",
  19,
  "buzz",
  "fizz",
  22,
  23,
  "fizz",
  "buzz",
  26,
  "fizz",
  28,
  29,
  "fizzBuzz",
  31,
  32,
  "fizz",
  34,
  "buzz",
  "fizz",
  37,
  38,
  "fizz",
  "buzz",
  41,
  "fizz",
  43,
  44,
  "fizzBuzz",
  46,
  47,
  "fizz",
  49,
  "buzz",
  "fizz",
  52,
  53,
  "fizz",
  "buzz",
  56,
  "fizz",
  58,
  59,
  "fizzBuzz",
  61,
  62,
  "fizz",
  64,
  "buzz",
  "fizz",
  67,
  68,
  "fizz",
  "buzz",
  71,
  "fizz",
  73,
  74,
  "fizzBuzz",
  76,
  77,
  "fizz",
  79,
  "buzz",
  "fizz",
  82,
  83,
  "fizz",
  "buzz",
  86,
  "fizz",
  88,
  89,
  "fizzBuzz",
  91,
  92,
  "fizz",
  94,
  "buzz",
  "fizz",
  97,
  98,
  "fizz",
  "buzz",
]
`;
